### PR TITLE
bugfix: encryption/decryption

### PR DESCRIPTION
**Pull request by nasaWelder**

### CODE REVIEW

File: merkato/merkato_config.py
Comment (mark-antony1) [inline merkato/merkato_config.py]: extract this to a util function

Comment (mark-antony1) [reply merkato/merkato_config.py]: *all of the if statements

File: merkato/merkato_config.py
Comment (mark-antony1) [inline merkato/merkato_config.py]: this seems mostly copy and pasted from decrypt_keys? Instead of supply the whole object, just supply the exchange config to decrypt keys. decrypt keys will mutate the config passed to it.

**Review by mark-antony1 — CHANGES_REQUESTED** [review]

File: merkato/merkato_config.py
Comment (nasaWelder) [reply merkato/merkato_config.py]: I don't like extracting stuff to other places, makes debugging a nightmare. We're still debugging way too much to be jumping across 5 files to trace a type issue. Edit: in this case, since it's the same thing in several places, sure. But in general this whole architecture scheme is not very OOP and hard to read. The DB and all its functions should be a flat object with no util scripts

File: merkato/merkato_config.py
Comment (nasaWelder) [reply merkato/merkato_config.py]: This is me doing that, then extracting it to here.

File: merkato/merkato_config.py
Comment (mark-antony1) [reply merkato/merkato_config.py]: I strongly disagree. Each function should be as small as possible, take as few arguments as possible and only do what would be expected from the function name. 

If the above is done properly, then function extraction is a must. This strategy makes things much more easy to read. This is best so you can see the stack trace of functions that describe their logic and literally read in English what was supposed to happen vs what did.

File: merkato/merkato_config.py
Comment (mark-antony1) [reply merkato/merkato_config.py]: The point is that decrypt merkato is duplicate code. The config for ONLY the exchange could be passed to decrypt_keys and accomplish the same result.

Comment (nasaWelder) [reply merkato/merkato_config.py]: You're right, I'm still getting used to the utility function thing editing arguments in place. I normally use object attributes, different way of thinking I guess

**Comment by needmoney90** [comment]
Is this mergeable, @nasaWelder? Has a small conflict, but doesn't look like much

**Comment by nasaWelder** [comment]
Just need to delete my change in that conflict

**Comment by nasaWelder** [comment]
@livinginformation should be good

**Comment by needmoney90** [comment]
@15chrjef waiting on an approve before merging

**Comment by nasaWelder** [comment]
added profit margin

**Comment by mark-antony1** [comment]
Hey @nasaWelder what here is still needed to be merged? Master is currently implemented in such a way that encryption and decryption work on simple_app

**Comment by nasaWelder** [comment]
Didn't work for me unless I added the bytes stuff. Also there's profit margin added to the GUI.

**Comment by nasaWelder** [comment]
Why isn't this merged yet?

**Comment by needmoney90** [comment]
@15chrjef is this mergeable?

**Comment by mark-antony1** [comment]
Nevermind, this is very much needed. It has a dependency on @livinginformation's pr with the merkato kill method. Nice work @nasaWelder

**Review by mark-antony1 — APPROVED** [review]

**Pull request merged**
